src/main/java/dev/ebrydeu/spring_boot_library/services/MessageService.java

```diff
@@ -0,0 +1,66 @@
+package dev.ebrydeu.spring_boot_library.services;
+import dev.ebrydeu.spring_boot_library.domain.dto.MessageDto;
+import dev.ebrydeu.spring_boot_library.domain.entities.Message;
+import dev.ebrydeu.spring_boot_library.exception.CustomExceptions;
+import dev.ebrydeu.spring_boot_library.repositories.MessageRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+    @Transactional
```

Comment:
Didn't watch Martin latest lecturers. What propose of `@Transactional` in service Layer ?

Reply:
I believe I read it here: https://medium.com/@bubu.tripathy/implementing-transactions-in-a-spring-boot-application-bc6b33e88557 
I dont remember Martin using it in his cat project :) But if its not needed for this project I will remove it :D

Reply:
By annotating methods with @Transactional we specify that the operations within those methods should be executed within a single transaction boundary. This means that either all the operations succeed and are committed to the database, or in case of any failure, all changes are rolled back (undone), ensuring data consistency.

Reply:
You can leave it. I was just curious.